### DISCUSSION TITLE
How to use Axios interceptors with SWR to revalidate access/refresh JWTs?

### DISCUSSION THREAD

**Discussion by chrber04** [vercel/swr#2206] [Q&A]
Hi, I was thinking about switching to SWR over just using axios but I stumbled upon an issue with the authentication of my app. It uses JWT authentication where the refresh token is stored as a httpOnly cookie on the client, and the access token is stored in memory on the client. 

When a request to an endpoint which requires authentication returns a 403 status, a response interceptor that has been applied to the axios instance attempts to refresh the users tokens using the refresh token. It also applies a new header to the request to imply it has already been sent, as to prevent an infinite loop if the server again returns a 403 status. If the server does return with the new access token, then we append that to the Authorization header on the original request and retry it.

Now my question is how would you implement this using SWR? Could you use the onErrorRetry option to achieve the same thing?

I have done my fair share of googling but I have yet to find a discussion on it, so I appreciate any input!

**Comment by HarleySalas** [comment]
did you ever find a nice solution for this? looking for the same. I'm usually using hooks and zustand to store the accessToken in memory to build my axiosPrivate fetcher, which wont work in this case, it seems.
